### PR TITLE
feat(launch-market): Add vault details to launch market receipt

### PR DESCRIPTION
**Pull request by jaredvu**
Screen Shot 2024-10-17 at 3 38 20 PM

Add vault hook to populate receipt data.

---

## Views

- `<NewMarketForm>`, `<NewMarketSelectionStep>`
  - use vault hooks to populate receipt data

- new `<MegaVaultYieldOutput>`
  - consolidate usage of vault apr and used in `<VaultInfoSections>`, `<VaultOverviewSection>`, `<LaunchMarketSidePanel>`, `<LaunchMarketDialog>`, `<NewMarketForm>`, `<NewMarketPreviewStep>`, `<NewMarketSelectionStep>`

## Constants

- `numbers`
  - Add `DEFAULT_VAULT_DEPOSIT_FOR_LAUNCH` for use throughout launchMarket components and views
 
## Hooks

- `hooks/vaultHooks`
  - Add `useVaultCalculationForLaunchingMarket` with hardcoded info for `vaultFormInfo`

---

### CODE REVIEW

File: src/views/forms/NewMarketForm/v7/NewMarketSelectionStep.tsx [hunk 1]
```diff
@@ -53,6 +55,8 @@ export const NewMarketSelectionStep = ({
   const isDisconnected = onboardingState === OnboardingState.Disconnected;
   const launchableMarkets = useLaunchableMarkets();
   const stringGetter = useStringGetter();
+  const vault = useLoadedVaultDetails().data;
+  const depositApr = vault?.thirtyDayReturnPercent && vault.thirtyDayReturnPercent / 100;
```
Comment (tyleroooo) [inline src/views/forms/NewMarketForm/v7/NewMarketSelectionStep.tsx:59]: I don't think we need to divide by 100, it's already ready to be passed into the Output.

File: src/views/forms/NewMarketForm/v7/NewMarketSelectionStep.tsx [hunk 2]
```diff
@@ -80,11 +84,16 @@ export const NewMarketSelectionStep = ({
                 <Output
                   type={OutputType.Percent}
                   tw="inline-block text-color-success"
-                  value={0.3456}
+                  value={depositApr}
```
Comment (tyleroooo) [inline src/views/forms/NewMarketForm/v7/NewMarketSelectionStep.tsx:87]: we are doing fraction digits = 0 for these

File: src/views/forms/NewMarketForm/v7/NewMarketPreviewStep.tsx [hunk 3]
```diff
@@ -48,6 +50,8 @@ export const NewMarketPreviewStep = ({
   const baseAsset = getDisplayableAssetFromTicker(ticker);
   const launchableAsset = useMetadataServiceAssetFromId(ticker);
   const { createPermissionlessMarket } = useSubaccount();
+  const vault = useLoadedVaultDetails().data;
+  const depositApr = vault?.thirtyDayReturnPercent && vault.thirtyDayReturnPercent / 100;
```
Comment (tyleroooo) [inline src/views/forms/NewMarketForm/v7/NewMarketPreviewStep.tsx:54]: no division, 0 fraction digits

File: src/views/dialogs/LaunchMarketDialog.tsx [hunk 4]
```diff
@@ -23,6 +25,8 @@ export const LaunchMarketDialog = ({
   const { isMobile } = useBreakpoints();
   const [formStep, setFormStep] = useState<NewMarketFormStep>();
   const stringGetter = useStringGetter();
+  const vault = useLoadedVaultDetails().data;
+  const depositApr = vault?.thirtyDayReturnPercent && vault.thirtyDayReturnPercent / 100;
```
Comment (tyleroooo) [inline src/views/dialogs/LaunchMarketDialog.tsx:29]: saame here

File: src/pages/portfolio/AccountOverviewSection.tsx [hunk 5]
```diff
@@ -52,7 +53,7 @@ export const MegavaultYieldTag = () => {
             <Output
               tw="mr-0.25"
               type={OutputType.Percent}
-              value={vault?.thirtyDayReturnPercent}
+              value={vault.thirtyDayReturnPercent / 100}
```
Comment (tyleroooo) [inline src/pages/portfolio/AccountOverviewSection.tsx:56]: pretty sure we don't divide

File: src/hooks/vaultsHooks.ts [hunk 6]
```diff
@@ -304,6 +304,40 @@ export const useVaultFormSlippage = () => {
   return mapNullableQueryResult(slippageQueryResult);
 };
 
+export const useVaultCalculationForLaunchingMarket = ({ amount }: { amount: number }) => {
+  const accountInfo = useAppSelector(selectSubaccountStateForVaults);
+  const slippageResponse = useVaultFormSlippage().data;
```
Comment (tyleroooo) [inline src/hooks/vaultsHooks.ts:309]: make this null

File: src/hooks/vaultsHooks.ts [hunk 7]
```diff
@@ -304,6 +304,40 @@ export const useVaultFormSlippage = () => {
...
+  const vaultAccount = useLoadedVaultAccount().data;
+
+  const vaultFormInfo = useMemo(
+    () => new VaultFormData(operationStringToVaultFormAction('DEPOSIT'), amount, true, true, true),
```
Comment (tyleroooo) [inline src/hooks/vaultsHooks.ts:313]: we probably do need them to acknowledge the terms
image

Comment (jaredvu) [reply src/hooks/vaultsHooks.ts:313]: Hmmm. The Launch Market flow has its own terms though

**Review by tyleroooo — APPROVED** [review]

**Review by tyleroooo — APPROVED** [review]

**Pull request merged**